### PR TITLE
Calories burnt

### PR DESCRIPTION
**Pull request by zhangerang**
calorie counter

### CODE REVIEW

File: src/js/views/classSession.js [hunk 1]
```diff
@@ -278,6 +283,31 @@ export default class ClassSession extends Component {
     );
   }
 
+  renderCalories() {
```
Comment (johnli611) [inline src/js/views/classSession.js]: move to separate component that has both calories and heart rate. call it something generic that indicates it's feedback/metrics for the user

File: src/js/views/classSession.js [hunk 2]
```diff
@@ -278,6 +283,31 @@ export default class ClassSession extends Component {
...
+    console.log(this.state.calories);
```
Comment (johnli611) [inline src/js/views/classSession.js]: remove all unnecessary console log statements

**Comment by zhangerang** [comment]
metrics counter

**Comment by zhangerang** [comment]
calorie counter

File: src/js/views/classSession.js [hunk 3]
```diff
@@ -144,6 +149,8 @@ export default class ClassSession extends Component {
   componentDidMount() {
     this.setInitialPoints();
     this.interval = setInterval(this.addPoints, 5000);
+    this.interval = setInterval(this.addCalories, 4500);
```
Comment (johnli611) [inline src/js/views/classSession.js]: are we sure it's okay to set multiple things to `this.interval`?

File: src/js/views/classSession.js [hunk 4]
```diff
@@ -144,6 +149,8 @@ export default class ClassSession extends Component {
...
+    this.interval = setInterval(this.addHeartRate, 5000);
```
Comment (johnli611) [inline src/js/views/classSession.js]: try updating heart rate 3500 ms

File: src/js/views/classSession.js [hunk 5]
```diff
@@ -278,6 +286,64 @@ export default class ClassSession extends Component {
     );
   }
 
+  renderMetrics(metric) {
```
Comment (johnli611) [inline src/js/views/classSession.js]: move to a separate component

File: src/js/views/classSession.js [hunk 6]
```diff
@@ -278,6 +286,64 @@ export default class ClassSession extends Component {
...
+    if(metric == 'calories') {
```
Comment (johnli611) [inline src/js/views/classSession.js]: space after `if`

File: src/js/views/classSession.js [hunk 7]
```diff
@@ -278,6 +286,64 @@ export default class ClassSession extends Component {
...
+    }
+  }
+
+  addCalories() {
+    if (!this.isCurrentlyStreaming()) {
+      return;
+    }
+
+    let addedCalories = Math.round(Math.random());
+    this.setState({
```
Comment (johnli611) [inline src/js/views/classSession.js]: space above

File: src/js/views/classSession.js [hunk 8]
```diff
@@ -278,6 +286,64 @@ export default class ClassSession extends Component {
     );
   }
 
+  renderMetrics(metric) {
+    if(metric == 'calories') {
+      return (
+        <div style={{margin: '10px', fontFamily:'Arimo', fontSize: '16px', display: 'inline-block'}} className="calories">
+          <img src={'https://s3-us-west-1.amazonaws.com/cloudworkout/fire+emoji.png'} style={{height: '15px', margin: '8px', marginBottom: '11px'}}/>
+          Calories
```
Comment (johnli611) [inline src/js/views/classSession.js]: add spaces between sibling-level divs/DOM elements

File: src/js/views/classSession.js [hunk 9]
```diff
@@ -278,6 +286,64 @@ export default class ClassSession extends Component {
...
+          <div style={{marginLeft: '30px', textAlign: 'center',fontFamily:'Arimo', fontSize: '28px'}}>
+            {this.state.calories}
+          </div>
+        </div>
+      )
+    } else if(metric == 'heart rate') {
+      return (
+        <div style={{margin: '10px', fontFamily:'Arimo', fontSize: '16px', display: 'inline-block'}} className="heart rate">
```
Comment (johnli611) [inline src/js/views/classSession.js]: what is the class `heart rate`? two separate classes?

File: src/js/views/classSession.js [hunk 10]
```diff
@@ -278,6 +286,64 @@ export default class ClassSession extends Component {
...
+      return (
+        <div style={{margin: '10px', fontFamily:'Arimo', fontSize: '16px', display: 'inline-block'}} className="calories">
+          <img src={'https://s3-us-west-1.amazonaws.com/cloudworkout/fire+emoji.png'} style={{height: '15px', margin: '8px', marginBottom: '11px'}}/>
+          Calories
+          <div style={{marginLeft: '30px', textAlign: 'center',fontFamily:'Arimo', fontSize: '28px'}}>
+            {this.state.calories}
+          </div>
+        </div>
+      )
+    } else if(metric == 'heart rate') {
```
Comment (johnli611) [inline src/js/views/classSession.js]: avoid spaces. use camelcase e.g. `heartRate` and set that as a const variable

File: src/js/views/classSession.js [hunk 11]
```diff
@@ -563,5 +574,96 @@ export default class ClassSession extends Component {
       });
     }
   }
+}
+
+//Metrics handles the rendering and changing values of the heart rate and calories
+class Metrics extends Component {
```
Comment (johnli611) [inline src/js/views/classSession.js]: every React class should be in its own file

**Pull request closed without merging**